### PR TITLE
distinguish system users

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

image

### CODE REVIEW

File: src/Search.php [hunk 1]
```diff
@@ -2516,6 +2527,7 @@ public static function showGenericSearch($itemtype, array $params)
         $p['is_deleted']   = 0;
         $p['as_map']       = 0;
         $p['browse']       = 0;
+        $p['is_sytem']     = 0;
```
Comment (flegastelois) [inline src/Search.php:2530]:
```suggestion
        $p['is_system']     = 0;
```

File: src/User.php [hunk 2]
```diff
@@ -2746,6 +2746,18 @@ public function showForm($ID, array $options = [])
                     'right'  => 'all'
                 ]);
                 echo "</td></tr>";
+
+                echo "<tr class='tab_bg_1'>";
+                echo "<td><label for='dropdown_profiles_id$grouprand'>" .  __('System user') . "</label></td><td>";
+                Html::showCheckbox([
```
Comment (cedric-anne) [inline src/User.php:2752]: I am not sure it is a good idea to permit to edit this field manually. IMHO, system users are meant to be used/updated only by system.

**Comment by cedric-anne** [comment]
I think that this feature will be really usefull only if we give ability to developer to "open a session" for a system user with full rights unlocked.
Indeed, I think that system users could be really usefull to be used in background contexts. For instance, we could automatically create a "cron system user" for GLPI and for all installed plugins, and switch to corresponding user before each crontask execution. With this, we would probably bypass some rights issues and we could probably improve log entries generated by system users.

File: src/CommonDBTM.php [hunk 3]
```diff
@@ -3188,6 +3188,20 @@ public function maybeDeleted()
         return array_key_exists('is_deleted', $this->fields);
     }
 
+    /**
+     * Is the object have system entries
+     *
+     * @return boolean
+     **/
+    public function haveSystemEntries()
```
Comment (btry) [inline src/CommonDBTM.php:3196]: I would suggest maybeSystem() to have a name similar to maybeDeleted()

File: src/User.php (hunk 2 above)
Comment (btry) [reply src/User.php:2752]: I agree with cedric-anne

File: src/Search/Input/QueryBuilder.php [hunk 4]
```diff
@@ -412,6 +412,7 @@ public static function displayCriteria($request = [])
             'criteria'   => $criteria ?? null,
             'prefix'     => $prefix,
             'p'         => $p,
+            'mainform'   => $p['mainform'] ?? 0,
```
Comment (orthagh) [inline src/Search/Input/QueryBuilder.php:415]: @cconard96 I'm not sure about this one.
There is a bug in list of users page where slider controls doesn't work anymore.
I found the mainform param was undefined when constructing first criteria.
Could you validate the fix is good ?

**Pull request closed without merging**